File: structural/facade/facade.java
public class FacadePattern {
    public static class Cat {
        private String name;

        public Cat(String name) {
            this.name = name;
        }

        public void jumpUp(String barrier) {
            String result = String.format("%s is on top of %s!", name, barrier);
            System.out.println(result);
        }

        public void jumpDown(String barrier) {
            String result = String.format("%s jumped down from %s!", name, barrier);
            System.out.println(result);
        }
    }

    public static class CatJumpsOverBarrierFacade {
        private Cat cat;

        public CatJumpsOverBarrierFacade(Cat cat) {
            this.cat = cat;
        }

        public void jumpOverBarrier(String barrier) {
            cat.jumpUp(barrier);
            cat.jumpDown(barrier);
        }
    }

    public static void main(String[] args) {
        Cat cat = new Cat("Simba");
        CatJumpsOverBarrierFacade catJumpsOverBarrierFacade = new CatJumpsOverBarrierFacade(cat);

        catJumpsOverBarrierFacade.jumpOverBarrier("fence");
    }
}

File: structural/facade/facade.cs
using System;

class FacadePattern
{
    class Cat
    {
        private string name { get; }

        public Cat(string name)
        {
            this.name = name;
        }

        public void JumpUp(string barrier)
        {
            Console.WriteLine("{0} is on top of {1}!", name, barrier);
        }

        public void JumpDown(string barrier)
        {
            Console.WriteLine("{0} jumped down from {1}!", name, barrier);
        }
    }

    class CatJumpsOverBarrierFacade
    {
        private Cat cat { get; }

        public CatJumpsOverBarrierFacade(Cat cat)
        {
            this.cat = cat;
        }

        public void JumpOverBarrier(string barrier)
        {
            cat.JumpUp(barrier);
            cat.JumpDown(barrier);
        }
    }

    static void Main(string[] args)
    {
        Cat cat = new Cat("Simba");
        CatJumpsOverBarrierFacade catJumpsOverBarrierFacade = new CatJumpsOverBarrierFacade(cat);

        catJumpsOverBarrierFacade.JumpOverBarrier("fence");
    }
}
